docs/seccompiler.md

```diff
@@ -0,0 +1,165 @@
+# Seccompiler - overview and user guide
+
+## Overview
+
+Seccompiler is a tool that compiles seccomp filters expressed as JSON files
+into serialized, binary BPF code that is directly consumed by Firecracker,
+at build or launch time.
+
+Seccompiler defines a custom [JSON file structure](#json-file-format), detailed
+further below, that the filters must adhere to.
+
+Besides the compiler binary, seccompiler also exports a small library
+interface, with a couple of helper functions, for deserializing and installing
+the binary filters.
+
+## Usage
+
+### Seccompiler binary
+
+To view the seccompiler command line arguments, pass the `--help` parameter to
+the executable.
+
+Example usage:
+
+```bash
+./seccompiler
+    --target-arch "x86_64"  # The CPU arch where the BPF program will run.
+                            # Supported architectures: x86_64, aarch64.
+    --input-file "x86_64_musl.json" # File path of the JSON input.
+    --output-file "bpf_x86_64_musl" # Optional path of the output file.
+                                    # [default: "seccomp_binary_filter.out"]
+```
+
+### Seccompiler library
+
+To view the library documentation, navigate to the seccompiler source code, in
+`firecracker/src/seccomp/src` and run `cargo doc --lib --open`.
+
+## Where is seccompiler implemented?
+
+Seccompiler is implemented as another package in the Firecracker cargo
+workspace. The code is located at `firecracker/src/seccomp/src`.
+
+## Supported platforms
+
+Seccompiler is supported on the
+[same platforms as Firecracker](../README.md#supported-platforms).
+
+## Release policy
+
+Seccompiler follows Firecracker's [release policy](RELEASE_POLICY.md) and
+version (it's released at the same time, with the same version number and
+adheres to the same support window).
+
+## JSON file format
+
+A JSON file expresses the seccomp policy for the entire Firecracker process. It
```

Comment:
Maybe it would be useful to put here a link to the default filter files so that users get a better idea of the format.